data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/PipelineConfigurationValidator.java

```diff
@@ -23,6 +25,7 @@ public class PipelineConfigurationValidator {
     private static final Logger LOG = LoggerFactory.getLogger(PipelineConfigurationValidator.class);
     private static final String PIPELINE_ATTRIBUTE_NAME = "name";
     private static final String PIPELINE_TYPE = "pipeline";
+    private static final Set<String> INVALID_PIPELINE_NAMES = new HashSet<>(List.of("data-prepper", "dataPrepper", "core"));
```

Comment:
This is to avoid conflicts in metrics prefix, if pipeline name is core then the metics from plugins will be `core.plugin-name.metricname` and core metrics will be `core.peerforwarder.metricname`. This is just to maintain the correct metric prefix.